### PR TITLE
Fixes #11726: Allow inventories to be augmented by script on the agent side

### PR DESCRIPTION
**Pull request by ncharles**
https://www.rudder-project.org/redmine/issues/11726

### CODE REVIEW

File: rudder-agent/SOURCES/patches/fusioninventory/11726-Extend-Rudder-specific-inventory-with-clie.patch [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+                             NAME NO_REMOVE RELEASE_TYPE PUBLISHER
+diff --git a/lib/FusionInventory/Agent/Task/Inventory/Generic/Rudder.pm b/lib/FusionInventory/Agent/Task/Inventory/Generic/Rudder.pm
+index b2fd04d..a36e450 100644
+--- a/lib/FusionInventory/Agent/Task/Inventory/Generic/Rudder.pm
++++ b/lib/FusionInventory/Agent/Task/Inventory/Generic/Rudder.pm
+@@ -7,6 +7,8 @@ use English qw(-no_match_vars);
+ 
+ use FusionInventory::Agent::Tools;
+ 
++use JSON::PP;
```
Comment (peckpeck) [inline rudder-agent/SOURCES/patches/fusioninventory/11726-Extend-Rudder-specific-inventory-with-clie.patch]: Do we have to install this as a new dependency ?

File: rudder-agent/SOURCES/patches/fusioninventory/11726-Extend-Rudder-specific-inventory-with-clie.patch [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+ 
++sub _getCustomProperties {
++    my (%params) = @_;
++    my $logger   = $params{logger};
++
++    my $custom_properties_dir = ($OSNAME eq 'MSWin32') ? 'C:\Program Files\Rudder\hooks.d' : '/var/rudder/hooks.d';
++    my $custom_properties;
++    if (-d "$custom_properties_dir") {
++        my @custom_propertes_list = ();
++        opendir(DIR, $custom_properties_dir); # or die $!;
```
Comment (peckpeck) [inline rudder-agent/SOURCES/patches/fusioninventory/11726-Extend-Rudder-specific-inventory-with-clie.patch]: We may need a "or return" here

File: rudder-agent/SOURCES/patches/fusioninventory/11726-Extend-Rudder-specific-inventory-with-clie.patch (hunk 1 above)
Comment (ncharles) [reply rudder-agent/SOURCES/patches/fusioninventory/11726-Extend-Rudder-specific-inventory-with-clie.patch]: no, this is already included and used by agent

**Comment by ncharles** [comment]
Commit modified

File: rudder-agent/SOURCES/patches/fusioninventory/11726-Extend-Rudder-specific-inventory-with-clie.patch [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+ 
++sub _getCustomProperties {
++    my (%params) = @_;
++    my $logger   = $params{logger};
++
++    my $custom_properties_dir = ($OSNAME eq 'MSWin32') ? 'C:\Program Files\Rudder\local\hooks.d' : '/var/rudder/local/hooks.d';
++    my $custom_properties;
++    if (-d "$custom_properties_dir") {
++        my @custom_propertes_list = ();
++        opendir(DIR, $custom_properties_dir); # or die $!;
```
Comment (peckpeck) [inline rudder-agent/SOURCES/patches/fusioninventory/11726-Extend-Rudder-specific-inventory-with-clie.patch]: We should do something if the open fails

**Comment by ncharles** [comment]
Commit modified

**Comment by ncharles** [comment]
Commit modified

**Pull request merged**